internal/wire/op_query.go

```diff
@@ -60,10 +61,18 @@ func (query *OpQuery) readFrom(bufr *bufio.Reader) error {
 	}
 
 	var q bson.Document
+
 	if err := q.ReadFrom(bufr); err != nil {
 		return err
 	}
-	query.Query = must.NotFail(types.ConvertDocument(&q))
+
+	doc := must.NotFail(types.ConvertDocument(&q))
+
+	if err := validateValue(doc); err != nil {
+		return NewValidationError(fmt.Errorf("wire.OpQuery.ReadFrom: validation failed for %v with: %v", doc, err))
```

Comment:
For clarity the fucntion's signature could be updated to highlight that a `ValidationError` could be returned.